src/jobsautoreport/report.py

```diff
@@ -1,31 +1,151 @@
+        )
+
+    def get_success_rate(self, from_date: datetime, to_date: datetime) -> float:
+        total_times = self.get_number_of_times_triggered(from_date, to_date)
+        successful_times = self._querier.query_number_of_times_by_state(
+            from_date, to_date, self.name, JobState.SUCCESS
+        )
+        if total_times == 0:
+            return 0
+        return (len(successful_times) / total_times) * 100
+
 
 class Reporter:
     def __init__(self, querier: Querier):
         self._querier = querier
 
-    def get_success_rate(self, from_date: datetime, to_date: datetime) -> str:
-        successful_jobs_list = self._querier.query_successful_jobs(from_date, to_date)
-        unsuccessful_jobs_list = self._querier.query_successful_jobs(from_date, to_date)
-        if len(successful_jobs_list) == 0:
-            return "0"
-        return f"{(len(successful_jobs_list) / (len(unsuccessful_jobs_list) + len(successful_jobs_list))) * 100}%"
+    def _get_number_of_e2e_jobs_by_type(
+        self, from_date: datetime, to_date: datetime, job_type: JobType
+    ) -> int:
+        job_list = self._querier.query_e2e_jobs_by_type(from_date, to_date, job_type)
+        return len(job_list)
 
-    def get_number_of_jobs_triggered(
-        self, from_date: datetime, to_date: datetime
+    def _get_number_of_e2e_jobs_by_type_and_state(
+        self,
+        from_date: datetime,
+        to_date: datetime,
+        job_type: JobType,
+        job_state: JobState,
     ) -> int:
-        jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
-        return len(jobs_list)
+        job_list = self._querier.query_e2e_jobs_by_job_type_and_state(
+            from_date, to_date, job_type, job_state
+        )
+        return len(job_list)
 
-    def get_average_duration_of_jobs(
-        self, from_date: datetime, to_date: datetime
-    ) -> Optional[timedelta]:
-        jobs_list = self._querier.query_number_of_jobs_triggered(from_date, to_date)
-        if len(jobs_list) == 0:
-            return None
-        avg = int(sum([job.duration for job in jobs_list]) / len(jobs_list))
-        return timedelta(seconds=avg)
+    def _get_success_rate_for_e2e_jobs_by_type(
+        self, from_date: datetime, to_date: datetime, job_type: JobType
+    ) -> float:
+        total_jobs = self._get_number_of_e2e_jobs_by_type(from_date, to_date, job_type)
+        successful_jobs = self._get_number_of_e2e_jobs_by_type_and_state(
+            from_date, to_date, job_type, JobState.SUCCESS
+        )
+        if total_jobs == 0:
+            return 0
+        return float(round((successful_jobs / total_jobs) * 100, 2))
+
+    def _get_top_n_e2e_jobs_by_type_and_state(
+        self,
+        from_date: datetime,
+        to_date: datetime,
+        job_type: JobType,
+        job_state: JobState,
+        n: int,
+    ) -> list[tuple[str, float]]:
+        res = []
+        jobs = self._querier.query_e2e_jobs_by_job_type_and_state(
```

Comment:
try to run the queries only once, and re-use the result to compute other metrics